Sources/Afluent/SequenceOperators/TimerSequence.swift

```diff
@@ -0,0 +1,84 @@
+//
+//  TimerSequence.swift
+//  Afluent
+//
+//  Created by Annalise Mariottini on 11/9/24.
+//
+
+import Foundation
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, visionOS 1.0, *)
+extension AsyncSequences {
+    /// A sequence that repeatedly emits a time duration offsetn on a given interval.
+    public struct TimerSequence<C: Clock>: AsyncSequence, Sendable {
+        public typealias Element = C.Instant
+
+        init(interval: C.Duration, tolerance: C.Duration?, clock: C) {
+            self.interval = interval
+            self.tolerance = tolerance
+            self.clock = clock
+        }
+
+        private let interval: C.Duration
+        private let tolerance: C.Duration?
+        private let clock: C
+
+        public struct AsyncIterator: AsyncIteratorProtocol {
+            private var cancellables: Set<AnyCancellable> = []
+
+            init(interval: C.Duration, tolerance: C.Duration?, clock: C) {
+                let (stream, continuation) = AsyncStream<C.Instant>.makeStream()
+                self.iterator = stream.makeAsyncIterator()
+
+                DeferredTask {
+                    while true {
+                        try await clock.sleep(for: interval, tolerance: tolerance)
+                        continuation.yield(clock.now)
+                    }
+                }.subscribe().store(in: &cancellables)
+            }
+
+            private var iterator: AsyncStream<Element>.Iterator
+
+            public mutating func next() async -> Element? {
+                await iterator.next()
+            }
+        }
+
+        public func makeAsyncIterator() -> AsyncIterator {
+            AsyncIterator(interval: interval, tolerance: tolerance, clock: clock)
+        }
+    }
+}
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, visionOS 1.0, *)
+public typealias TimerSequence = AsyncSequences.TimerSequence
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, visionOS 1.0, *)
+extension TimerSequence where C == ContinuousClock {
+    /// Returns a sequence that repeatedly emits the current offset of a continuous clock on the given interval.
```

Comment:
Nit it emits the instant. An instant may in practice be an offset from a reference for most clocks but I don't think that with the generic abstraction that's necessarily true.